js/curation-tool.js

```diff
@@ -892,23 +902,81 @@ const vm = new Vue({
       // from computer to computer. However, they are specific to the display
       // on which the phylogeny was curated and are otherwise unrelated to the
       // phylogeny itself. Therefore, I'm storing them elsewhere in the Vue
-      // model -- users will need to set the scaling every time they open this
+      // model -- users will need to set the spacing every time they open this
       // PHYX file.
-      if (!this.hasProperty(this.phylogenySpacingX, phylogeny)) {
-        Vue.set(this.phylogenySpacingX, phylogeny, 20);
+
+      // Phylogeny-specific spacing is stored by the index number of the phylogeny.
+      const phylogenyIndex = this.testcase.phylogenies.indexOf(phylogeny);
+
+      // If either spacingX or spacingY is not already set, we set them to their
+      // defaults.
+      if (!this.hasProperty(this.phylogenySpacingX, phylogenyIndex)) {
+        Vue.set(this.phylogenySpacingX, phylogenyIndex, DEFAULT_SPACING_X);
       }
-      if (!this.hasProperty(this.phylogenySpacingY, phylogeny)) {
-        Vue.set(this.phylogenySpacingY, phylogeny, 40);
+      if (!this.hasProperty(this.phylogenySpacingY, phylogenyIndex)) {
+        Vue.set(this.phylogenySpacingY, phylogenyIndex, DEFAULT_SPACING_Y);
       }
 
+      // Place and update the nodes on the phylogeny.
       tree
-        .spacing_x(this.phylogenySpacingX[phylogeny])
-        .spacing_y(this.phylogenySpacingY[phylogeny])
+        .spacing_x(this.phylogenySpacingX[phylogenyIndex])
+        .spacing_y(this.phylogenySpacingY[phylogenyIndex])
         .placenodes()
         .update();
       return tree;
     },
 
+    // Get and set phylogeny spacing information.
+    getPhylogenySpacingX(phylogenyIndex) {
+      // Return the phylogeny spacing in the X dimension for a particular phylogeny by index.
+      // Will set up the spacing variable if it has not been created yet, so this should be
+      // called before accessing this.phylogenySpacingX directly.
+      if (!this.hasProperty(this.phylogenySpacingX, phylogenyIndex)) {
+        Vue.set(this.phylogenySpacingX, phylogenyIndex, DEFAULT_SPACING_X);
+      }
+
+      return this.phylogenySpacingX[phylogenyIndex];
+    },
+
+    changePhylogenySpacingX(phylogenyIndex, changeBy) {
+      // Change the phylogeny spacing in the X dimension by the changeBy value.
+      // We prevent the value from going below MIN_SPACING_X.
+      if (!this.hasProperty(this.phylogenySpacingX, phylogenyIndex)) {
+        Vue.set(this.phylogenySpacingX, phylogenyIndex, DEFAULT_SPACING_X);
+      }
+
+      this.phylogenySpacingX[phylogenyIndex] += changeBy;
```

Comment:
The subsequent conditional adjustment shows that this line may be setting the value of the property to a disallowed or unacceptable value. This may in reality be entirely inconsequential right now because the adjustment to be within acceptable limits follows immediately, i.e., probably at most within a few milliseconds. Note however that it's not defensive programming: if you had someone watching the object property, they'd see a flicker into disallowed territory. It's easy to avoid by just using a temporary variable:
```suggestion
      let spacing = this.phylogenySpacingX[phylogenyIndex] + changeBy;
      this.phylogenySpacingX[phylogenyIndex] = spacing < MIN_SPACING_X ? MIN_SPACING_X : spacing;
```

Reply:
It seems BTW that "Suggested change" doesn't let me indicate that this should replace several lines, so you'll have to edit it to delete the following if condition if you want to accept it 😃